verification/tests/solidity_multiple.rs

```diff
@@ -0,0 +1,389 @@
+        test_success(contract_dir, test_input).await;
+    }
+
+    #[actix_rt::test]
+    async fn verifies_the_generated_bytecode_with_external_libraries() {
+        let contract_dir = "contract_with_lib";
+        let mut libraries = BTreeMap::new();
+        libraries.insert(
+            "BadSafeMath".to_string(),
+            "0x9Bca1BF2810c9b68F25c82e8eBb9dC0A5301e310".to_string(),
+        );
+        // let test_input = TestInput::new("SimpleStorage", "v0.5.11+commit.c082d0b4")
+        let test_input = TestInput::new("SimpleStorage", "v0.5.11+commit.22be8592")
+            .with_optimization_runs(200)
+            .with_contract_libraries(libraries);
+        test_success(contract_dir, test_input).await;
+    }
+
+    #[actix_rt::test]
+    // `whisper` metadata - (bzz0 => bzz1) in solidity 0.5.11()
+    async fn verifies_smart_contract_with_new_whisper_metadata() {
+        let contract_dir = "solidity_5.11_new_whisper_metadata";
+        let test_input = TestInput::new("FixedSupplyToken", "v0.5.11+commit.22be8592")
+            .with_evm_version("byzantium");
+        test_success(contract_dir, test_input).await;
+    }
+
+    #[actix_rt::test]
+    async fn verifies_library() {
+        let contract_dir = "library";
+        // let test_input = TestInput::new("Foo", ""v0.5.11+commit.c082d0b4"")
+        let test_input =
+            TestInput::new("Foo", "v0.5.11+commit.22be8592").with_optimization_runs(200);
+        test_success(contract_dir, test_input).await;
+    }
+
+    #[actix_rt::test]
+    // (includes new metadata in bytecode)
+    async fn verifies_smart_contract_compiled_with_solidity_0_5_9() {
+        let contract_dir = "solidity_0.5.9_smart_contract";
+        // let test_input = TestInput::new("TestToken", "v0.5.9+commit.e560f70d")
+        let test_input = TestInput::new("TestToken", "v0.5.9+commit.c68bc34e")
+            .with_evm_version("petersburg")
+            .has_constructor_args();
+        test_success(contract_dir, test_input).await;
+    }
+
+    #[actix_rt::test]
+    // https://solidity.readthedocs.io/en/v0.6.6/contracts.html?highlight=immutable#constant-and-immutable-state-variables
+    async fn verifies_smart_contract_with_immutable_assignment() {
+        let contract_dir = "with_immutable_assignment";
+        let test_input = TestInput::new("C", "v0.6.7+commit.b8d736ae")
+            .with_optimization_runs(200)
+            .has_constructor_args();
+        test_success(contract_dir, test_input).await;
+    }
+
+    #[actix_rt::test]
+    // verifies smart-contract created from another contract
+    async fn contract_from_factory() {
+        let contract_dir = "contract_from_factory";
+        let test_input = TestInput::new("ContractFromFactory", "v0.4.26+commit.4563c3fc")
+            .with_optimization_runs(200)
+            .has_constructor_args();
+        test_success(contract_dir, test_input).await;
+    }
+}
+
+mod basic_error_tests {
```

Comment:
I would also add test with wrong compiler version, but it's not super important right now